ls1intum/Artemis
# [Enhancement] - Improve regex in re-evaluate

### Checklist

### Motivation and Context


The regex in the short answer question edit component did not handle spots correctly which are directly connected to another word. For example
 - `[-spot 10];System.out.println();` was not split after `]`
 - `[index + 7]` was treated as spot
 - `myArray.[-spot 4]` was not split before `[`
This caused adjacent strings to disappear in the visual mode, which re-evaluate uses.

### Description

We improved the regex, which now handles also these situations

### Steps for Testing


1. Log in to Artemis
2. Create a short answer question exercise
3. Remove the whitespace between a spot and the following word
4. participate
5. click on re-evaluate
6. make sure the word following the spot is correctly displayed in the view

### Screenshot
For example a question like 

> [-spot 20].out.prinltn();

Results in:
![Screenshot 2021-03-11 at 18 16 12](https://user-images.githubusercontent.com/44401560/110827908-ccf14000-8296-11eb-8ea7-a713481b7e3a.png)
